### DISCUSSION TITLE
installazione zigbee2mqtt in Home assistant OS

### DISCUSSION THREAD

**Discussion by efestopini** [indomus/forum#2803] [Networking]
salve,
sto seguendo questa guida per integrare zigbee2mqtt
 https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-home-assistant-hassio/ 
potreste gentilmente dirmi dove sbaglio, sono arrivato all’installazione del’ add-on zigbee2mqtt, cliccato su configurazione, tre puntini e modifica in YAML, a questo punto la guida mostra come dovrebbe presentarsi il file e che i campi da personalizzare sono, principalmente:

"homeassistant";
"server";
"user" e "password";
"serial".

ma il mio problema è non ci sono quei campi, io ho solo 9 righe

data_path: /config/zigbee2mqtt
socat:
enabled: false
master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
options: "-d -d"
log: false
mqtt: {}
serial: {}

praticamente ho solo il campo serial, quelle che non ci sono le devo scrivere per intero? Ho scritto questa domanda anche sotto la guida ma forse è meglio chiedere qui

**Comment by KC-inDomus** [comment]
https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-home-assistant-hassio/#config

c'è tutta la configurazione completa, applicala

**Reply by efestopini** [reply]
grazie mille, è che essendo la prima volta che mi cimento in queste cose, ogni piccola cosa diventa un dubbio enorme, per esempio la guida dice che aprendo il file troverò una configurazione di default pressapoco così, io apro il file è non trovando scritto praticamente niente vado nel panico. Copio tutto, modifico quello che è da modificare e procedo, giusto per sicurezza, aggiungo il tutto prima di quello che già trovo scritto o cancello e metto solo quello scritto nella guida? grazie ancora

**Reply by efestopini** [reply]
non capisco, io copio il tutto, mettendo l'ip del pc dove gira HASSIO e il broker mqtt, metto l' utente e la password, metto il seriale che penso sia giusto, ma mi da questi errori
![qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq](https://github.com/indomus/forum/assets/154065991/ff765b9e-8c40-4869-98b1-e8b655b404ae)

**Reply by efestopini** [reply]
problema risolto, purtroppo la guida qui su indomus deve essere un po vecchiotta, niente di problematico per chi si sa già muovere ma spiazzante per chi si approccia per la prima volta praticamente ora la configurazione è divisa in sezioni 
![wwwwwwwwwwwwwwwwwwwwwwwww](https://github.com/indomus/forum/assets/154065991/e7cda0da-b178-49ee-9caa-809ee986f0b3)

**Reply by KC-inDomus** [reply]
c'è scritto 

"Tale configurazione si effettua sempre dalla pagina di dettaglio dell’add-on ZigBee2MQTT; presso la sezione “Configurazione“, cliccando poi sui tre puntini in alto a destra, infine alla voce “Modifica in YAML“, l’add-on riporta una configurazione di default pressapoco così:"

certo che se ti limiti a usare la gui, è normale che trovi delle differenze. ciò detto daremo un'occhiata, perchè c'è porzione di codice che cmq non serve.

**Reply by efestopini** [reply]
scusami tanto se no capisco, lo screen che ho messo è appunto della sezione "configurazione", solo che probabilmente all' epoca della guida esisteva un unico file YAML quindi andava scritto tutto nello stesso posto, ora invece è diviso in sezioni e se si scrive il tutto nella prima sezione socat (che è quella che si apre cliccando i tre puntini, viene fuori pieno di errori.
L'importante è che ne sono venuto a capo, non capisco anche cosa intendi per "certo che se ti limiti a usare la gui", io cerco (tra le mie mille difficoltà) di seguire scrupolosamente la guida, se questa mi dive pigia li, scrivi questo la, ecc.. io faccio quello, se esistono altri modi io non li conosco di sicuro, i miei problemi escono quando per via di aggiornamenti cambiano le posizioni o nomi di comandi 

ora la schermata è così
![image](https://github.com/indomus/forum/assets/154065991/19f64823-5579-4016-afc2-e34b9fe52790)

ovviamente non vuole essere una lamentela, mi rendo conto che non si può mica stare dietro ad aggiornare ogni guida per ogni cambiamento che avviene, sarebbe un lavoro a tempo pieno
